app/models/form/sales/pages/deposit.rb

```diff
@@ -0,0 +1,21 @@
+class Form::Sales::Pages::Deposit < ::Form::Page
+  def initialize(id, hsh, subsection, ownershipsch:, optional:)
+    super(id, hsh, subsection)
+    @ownershipsch = ownershipsch
+    @optional = optional
+    @header = "About the deposit"
+  end
+
+  def questions
+    @questions ||= [
+      Form::Sales::Questions::DepositAmount.new(nil, nil, self, ownershipsch: @ownershipsch, optional: @optional),
+    ]
+  end
+
+  def routed_to?(log, _user)
+    return true if log.ownershipsch == 2 || (log.ownershipsch == 3 && log.mortgageused == 1)
+    return false if log.stairowned_100? != @optional && form.start_year_after_2024?
+
+    log.social_homebuy? == false && log.ownershipsch == 1 || log.social_homebuy? == true
```

Comment:
Is social_homebuy relevant if the log is not shared ownership? Looks like the options are based on type, and I think the type options are fairly different based on ownership scheme.

Reply:
Yeah, you're right 👀 removed that and it still passes the tests